### DISCUSSION TITLE
How to use `code-server --link` in WSL

### DISCUSSION THREAD

**Discussion by lg-nineoneninethreenine** [coder/code-server#3966] [Q&A]
I use WSL for code-server. How can I enable the `--link` flag? It will fail whenever I try it so how can I do it?

PS: I use Ubuntu ***AND*** Debian WSL

**Comment by lg-nineoneninethreenine** [comment]
Moreover, how can I get the domain that comes in `code-server --link`? And how can I activate it for code-server?

**Reply by jsjoeio** [reply]
Hmm...I don't have WSL to test 🤔 cc @code-asher 

> how can I get the domain that comes in code-server --link

I think the format is something like `https://<computer name?>-<github handle>.cdr.co`. In my scenario its "https://joes_mbp-jsjoeio.cdr.co" but I'm not sure where it gets "joes_mbp" 🤔

**Reply by code-asher** [reply]
I think it uses your machine's host name by default if you don't provide an argument to `--link`.

**Reply by lg-nineoneninethreenine** [reply]
But, how in the whole wide world do I enable it? I tried it and every time it fails. I will try putting the Ubuntu log here soon.

**Reply by lg-nineoneninethreenine** [reply]
> Hmm...I don't have WSL to test 🤔 cc @code-asher 
> 
> > how can I get the domain that comes in code-server --link
> 
> I think the format is something like `https://<computer name?>-<github handle>.cdr.co`. In my scenario its "https://joes_mbp-jsjoeio.cdr.co" but I'm not sure where it gets "joes_mbp" 🤔

@jsjoeio I tried visiting the link you put there and it asked me to sign in to Coder Cloud. How can I get that kind of link of sorts?

**Reply by lg-nineoneninethreenine** [reply]
@jsjoeio you use a predefined domain. How did you manage to get that?

**Reply by jsjoeio** [reply]
You should be able to run `code-server --link guacamole` locally and it spits out a link in the logs, you open in your browser and sign in and that's it. If it's not working then I'm not sure 🤔 

Here's a video: https://www.youtube.com/watch?v=MdIEiZpg5vQ

**Reply by lg-nineoneninethreenine** [reply]
It ain't workin'. Check this:

https://github.com/cdr/code-server/discussions/3966#discussioncomment-1166477

At the same discussion btw

**Comment by code-asher** [comment]
In what way does it fail? Are there any errors?

**Reply by lg-nineoneninethreenine** [reply]
Well, this:
```bash
[2021-08-13T04:22:06.392Z] info  code-server 3.10.2 387b12ef4ca404ffd39d84834e1f0776e9e3c005
[2021-08-13T04:22:06.395Z] info  Using user-data-dir ~/.local/share/code-server
[2021-08-13T04:22:06.447Z] info  Using config file ~/.config/code-server/config.yaml
[2021-08-13T04:22:06.448Z] info  HTTP server listening on http://127.0.0.1:35989 (randomized by --link)
[2021-08-13T04:22:06.449Z] info    - Authentication is disabled (disabled by --link)
[2021-08-13T04:22:06.449Z] info    - Not serving HTTPS (disabled by --link)
[2021-08-13T04:22:09.711Z] info  Unfortunately we cannot ensure a good user experience with your connection latency (139ms). Efforts are underway to accommodate users in most areas.
[2021-08-13T04:22:09.713Z] error --link agent exited with 1
```
I have one of the fastest network lines in my area but it says it cannot accommodate my latency. What can I do about this?

**Reply by lg-nineoneninethreenine** [reply]
@code-asher just did the SAME thing on Gitpod Online IDE and it accepted the latency of 51 ms. Is there any way to control network latency on a PC?

**Reply by jsjoeio** [reply]
> Unfortunately we cannot ensure a good user experience with your connection latency (139ms). Efforts are underway to accommodate users in most areas.

Hmmm 🤔 I think we've seen the same thing mentioned before. I'm not sure how latency is calculated but maybe @code-asher knows

**Reply by code-asher** [reply]
I haven't seen the code but I suspect it's just using something like a ping to our cloud.

**Reply by lg-nineoneninethreenine** [reply]
It is supposed to be below 75 ms. And my network line is one of the most fastest in my area. What is making ***this*** happen?

**Reply by lg-nineoneninethreenine** [reply]
I think I must call myself a fool. I have to upgrade to 3.11.1 firstly.

**Reply by lg-nineoneninethreenine** [reply]
Moreover, today, I was in a Google Meet meeting and I happened to see the latency. Below 75ms most of the time. Is it problems with PC or WSL?

**Reply by code-asher** [reply]
If the ping to our cloud is too high then link won't let you connect unfortunately.

But we're currently rewriting link to work peer to peer and I think it should resolve the issue once we're finished!

**Reply by lg-nineoneninethreenine** [reply]
> If the ping to our cloud is too high then link won't let you connect unfortunately.
> 
> But we're currently rewriting link to work peer to peer and I think it should resolve the issue once we're finished!

Okay, so till then, I can't use link, right?

**Reply by lg-nineoneninethreenine** [reply]
Why not add the common issues for the link flag in the docs? When people read that, they will have an idea on how the error occurs, right?

**Reply by code-asher** [reply]
> Okay, so till then, I can't use link, right?

Yes, that is right.

> Why not add the common issues for the link flag in the docs?

I agree!

**Reply by lg-nineoneninethreenine** [reply]
Yup, same. Put it in a page called Public issues and put every public code-server issue there. Then, if it gets fixed, you can remove it and mark it as solved.
